### ISSUE TITLE
Load balancer ports published incorrectly for local connectivity

### ISSUE THREAD

**Issue by mithrandi** [rancher/rancher#7349]
**Rancher Versions:**
Server: 1.3.0
healthcheck: 0.2.0
ipsec: 0.8.1
network-services: rancher/metadata:v0.6.8, rancher/network-manager:v0.4.0
scheduler: 0.5.0

**Docker Version:**
1.11.2

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle

**Steps to Reproduce:**
Create a load balancer with a host IP specified.

**Results:**
Connections from containers on that Rancher host to host IPs other than the one specified still get directed to the load balancer.

I believe the explanation for this lies in the iptables rules installed for the load balancer:

```
-A CATTLE_OUTPUT -p tcp -m tcp --dport 443 -m addrtype --dst-type LOCAL -j DNAT --to-destination 10.42.115.20:443
-A CATTLE_PREROUTING -d 197.189.229.122/32 ! -i docker0 -p tcp -m tcp --dport 443 -j MARK --set-xmark 0x1068/0xffffffff
-A CATTLE_PREROUTING -d 197.189.229.122/32 ! -i docker0 -p tcp -m tcp --dport 443 -j DNAT --to-destination 10.42.115.20:443
-A CATTLE_PREROUTING -d 197.189.229.122/32 -p tcp -m tcp --dport 443 -j DNAT --to-destination 10.42.115.20:443
```
In this example, 197.189.229.122 is the host IP that I specified when setting up the load balancer. Note how the `CATTLE_PREROUTING` rules have a destination IP address selector, so connections from outside of the host work as expected; however, the `CATTLE_OUTPUT` rule does not have a destination IP address selector, so it intercepts all outgoing connections to any local IP address on port 443. Since I am in fact running the Rancher server on the same host (bound to a different IP address), this causes the agent to disconnect as now its attempts to reach the Rancher server are instead hitting the load balancer. Removing the `CATTLE_OUTPUT` rule restores connectivity until the rule is replaced by the Rancher networking stack.

**Comment by aemneina** [comment]
@mithrandi I think you nailed down the root cause, using 443 on a host that houses both server and agent, then starting up a service binding to that port causing network connectivity 'breakage'. This is a non-production use case (server and agent on the same host) so support for this kind of use case would be fringe IMO. @ibuildthecloud what say you?

**Comment by aemneina** [comment]
@mithrandi might have spoken too soon, it looks like this might solve your issue: #7034 Slated for rancher v1.4.0. If you agree it'll meet the need, we can close this.

**Comment by mithrandi** [comment]
Hmm, #7034 doesn't seem like it would _directly_ address my problem, since I don't actually need Rancher to schedule on multiple IP addresses, but I guess this issue might be addressed at the same time. I should note that my setup worked fine prior to Rancher 1.2.0 as well, the way ports were published on the load balancer in 1.1.x and earlier did not interfere with local connections in this way.

**Comment by mithrandi** [comment]
Still not working in 1.4, FWIW. I believe this is the code responsible for these iptables rules:

https://github.com/rancher/plugin-manager/blob/9abb1192dbcb8f6360a332eeb03eff9277383122/hostports/watcher.go#L93

There is no logic at all around the destination IP address, only the `--dst-type LOCAL` which will match any local traffic, not only traffic to the configured IP address.

**Comment by yuriiashutkin** [comment]
1.6.14 Not working. 
If I set host IP I got an error: 
```Failed to allocate instance [container:1i529]: Bad instance [container:1i529] in state [stopping]: Allocation failed: No healthy hosts meet the resource constraints: [172.17.0.1:2368:2368/tcp portReservation, instanceReservation: 1].```
Even with 10.42.0.1 and 127.0.0.1 (this one possible should not working)
It's only work if I leave Host IP field empty. And there is a problem. I don't want:
1) Grant access from everyone to external port
2) I don't want lb listen on any IP addresses I have on my server (there is 3 external IP addresses)

Also I can't find out how to prevent access to this shared to whole internet port via iptables. Even
```iptables -I INPUT 1 -p tcp --dport 2368 -j DROP``` not working. I'm sad and disappointed.
